core/src/test/java/org/apache/iceberg/avro/AvroTestHelpers.java

```diff
@@ -78,6 +79,18 @@ static void assertEquals(Types.StructType struct, Record expected, Record actual
     }
   }
 
+  static void assertEquals(Types.StructType struct, StructLike expected, StructLike actual) {
+    List<Types.NestedField> fields = struct.fields();
+    for (int i = 0; i < fields.size(); i += 1) {
+      Type fieldType = fields.get(i).type();
+
+      Object expectedValue = expected.get(i, Object.class);
+      Object actualValue = actual.get(i, Object.class);
```

Comment:
Instead of `Object.class`, this should use `fieldType.typeId().javaClass()` because the internal object model uses those types.